app/main.py

```diff
@@ -0,0 +1,137 @@
+from typing import Type
+from copy import deepcopy
+
+from fastapi import FastAPI, HTTPException
+from pydantic import BaseModel, Field, UUID4
+
+from models import Secret
+
+
+app = FastAPI()
+
+
+@app.get("/")
+def read_root():
+    return {"Hello": "World"}
+
+
+IN_MEMORY_DB = dict()
+
+
+class NotFoundException(Exception):
+    def __init__(self):
+        super().__init__("Secret with provided ID does not exist")
+
+
+class NotCreatedException(Exception):
+    def __init__(self):
+        super().__init__("Error creating secret")
+
+
+class SecretRepository:
```

Comment:
Please separate Repository pattern in a repository.py inside models and also the IN_MEMORY_DB goes there